File: Core-backend-layer/Programming-fundamentals/Java/UnitTesting/src/main/java/org/example/Product.java
package org.example;

public class Product {
    private String name;
    private double price;

    public Product(String name, double price) {
        this.name = name;
        this.price = price;
    }

    public String getName() {
        return name;
    }

    public double getPrice() {
        return price;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    // Display product information
    public void displayProductInfo() {
        System.out.println("Product: " + name + ", Price: $" + price);
    }

    // Method to apply a discount to the product price
    public void applyDiscount(double discountPercentage) {
        if (discountPercentage < 0 || discountPercentage > 100) {
            throw new IllegalArgumentException("Discount percentage must be between 0 and 100.");
        }
        double discountedPrice = price - (price * discountPercentage / 100);
        setPrice(discountedPrice);
    }
}

File: Core-backend-layer/Programming-fundamentals/Python/UnitTesting/src/main/product.py
class Product:
    def __init__(self, name, price):
        self.name = name
        self.price = price

    def get_name(self):
        return self.name

    def get_price(self):
        return self.price

    def set_name(self, name):
        self.name = name

    def set_price(self, price):
        self.price = price

    def display_product_info(self):
        print(f"Product: {self.name}, Price: ${self.price:.2f}")

    def apply_discount(self, discount_percentage):
        if discount_percentage < 0 or discount_percentage > 100:
            raise ValueError("Discount percentage must be between 0 and 100.")
        discounted_price = self.price - (self.price * discount_percentage / 100)
        self.set_price(discounted_price)